### ISSUE TITLE
After upgrade to v2.3.0, Ffmpeg process for webcam does not terminate (Python 2.7)

### ISSUE THREAD

**Issue by dennisvang** [imageio/imageio#343]
Description
-----------

After upgrading from imageio v2.2.0 to v2.3.0 (on a Windows system with Python 2.7 environment), it appears that the ffmpeg process is no longer terminated properly after closing a webcam Reader instance.

How to reproduce issue
------------------------

Here's a minimal test:

```python
import psutil
import imageio

def ffmpeg_alive():
    """ enumerate ffmpeg processes, then wait for them to terminate """
    ffmpeg_processes = []
    for process in psutil.process_iter():
        try:
            if 'ffmpeg' in process.name().lower():
                ffmpeg_processes.append(process)
        except psutil.NoSuchProcess as e:
            print e.message
    still_alive = False
    if ffmpeg_processes:
        __, still_alive = psutil.wait_procs(ffmpeg_processes, timeout=1)
    return still_alive


def test_webcam_ffmpeg_process_termination():
    """ open reader for a webcam, using context manager, then check ffmpeg process status """
    with imageio.get_reader('<video0>') as reader:
        assert reader._proc is not None
        assert ffmpeg_alive()
    assert reader._proc is None
    assert not ffmpeg_alive()


# Run test
print 'imageio version {}'.format(imageio.__version__)
test_webcam_ffmpeg_process_termination()
```

On my system this test *passes* when using a clean conda env running python 2.7 and imageio v2.2.0, but it *fails* when running an (otherwise identical) conda env with imageio v2.3.0 (environment details below). 

Probable cause
---------------

The culprit appears to be the `shell=ISWIN` argument in the `sp.Popen` call in `ffmpeg.Reader._initialize`. 

Proposed solution
-------------------
Quick fix for my specific case would be to remove the `shell` argument (i.e. revert that line to v2.2.0). However, the argument was probably added with a reason? 

Otherwise, a better option would probably be to modify the `ffmpeg.Reader._terminate` method, replacing `self._proc.kill()` by something like the following:

```python
if self.request._video:
    self._proc.communicate('q')
else:
    self._proc.kill()
```

At least this works on my system.

Environment details
--------------------

v2.2.0 environment:

```
certifi                   2016.2.28                py27_0
imageio                   2.2.0                     <pip>
numpy                     1.14.3                    <pip>
Pillow                    5.1.0                     <pip>
pip                       9.0.1                    py27_1
psutil                    5.2.2                    py27_0
python                    2.7.13                        1
setuptools                36.4.0                   py27_1
vc                        9                             0
vs2008_runtime            9.00.30729.5054               0
wheel                     0.29.0                   py27_0
wincertstore              0.2                      py27_0

```

v2.3.0 environment:

```
certifi                   2016.2.28                py27_0
imageio                   2.3.0                     <pip>
numpy                     1.14.3                    <pip>
Pillow                    5.1.0                     <pip>
pip                       9.0.1                    py27_1
psutil                    5.2.2                    py27_0
python                    2.7.13                        1
setuptools                36.4.0                   py27_1
vc                        9                             0
vs2008_runtime            9.00.30729.5054               0
wheel                     0.29.0                   py27_0
wincertstore              0.2                      py27_0

```

System is Windows 10 64 bit, using PyCharm professional 2018.1. Not sure if it matters, but for legacy reasons we are using Miniconda 32-bit.

**Comment by almarklein** [comment]
@dennisvang Thanks for this report!

The `shell=ISWIN` was added in #268 to fix #266 (cc @paulmueller)

In short: using shell mode on Windows was added to prevent a window popping up when frozen, but it prevents closing ffmpeg. I hope we can fix this by updating the termination function, otherwise we must do something like `shell=ISWIN and ISFROZEN`.

 @dennisvang How did you come up with `self._proc.communicate('q')`? If that indeed works, could you perhaps prepare a PR for that?

**Comment by paulmueller** [comment]
I recently stumbled across a similar problem with imageio. I got a lot of "OSError: [WinError 6] The handle is invalid" when running pytest. The problem was that I was not using `imageio.get_reader()` together with the `with` statement (due to performance reasons).

A workaround for me was to call  `cap._proc.kill()` twice. I found that by accident and have no idea why it works, but maybe someone with more insight does.

short version:
```python
cap = imageio.get_reader(path)
# [do something with cap]
cap._proc.kill()
cap.close()   # also calls _proc.kill()
```

long version:
https://github.com/ZELLMECHANIK-DRESDEN/dclab/blob/master/dclab/rtdc_dataset/fmt_tdms/event_image.py#L114-L123
